src/urllib3/response.py

```diff
@@ -221,35 +240,35 @@ def data(self) -> bytes:
         raise NotImplementedError()
 
     @property
-    def url(self) -> str:
+    def url(self) -> Optional[str]:
         raise NotImplementedError()
 
     @property
     def closed(self) -> bool:
         raise NotImplementedError()
 
     @property
-    def connection(self):
+    def connection(self) -> Optional[HTTPConnection]:
         raise NotImplementedError()
 
     def stream(
-        self, amt: int = 2 ** 16, decode_content: typing.Optional[bool] = None
-    ) -> typing.Generator[bytes, None, None]:
+        self, amt: int = 2 ** 16, decode_content: Optional[bool] = None
+    ) -> Iterator[bytes]:
         raise NotImplementedError()
 
     def read(
         self,
-        amt: typing.Optional[int] = None,
-        decode_content: typing.Optional[bool] = None,
+        amt: Optional[int] = None,
+        decode_content: Optional[bool] = None,
         cache_content: bool = False,
-    ) -> bytes:
+    ) -> Optional[bytes]:
         raise NotImplementedError()
 
     def read_chunked(
         self,
-        amt: typing.Optional[int] = None,
-        decode_content: typing.Optional[bool] = None,
-    ) -> bytes:
+        amt: Optional[int] = None,
+        decode_content: Optional[bool] = None,
+    ) -> Generator[bytes, None, None]:
```

Comment:
Can we do `Iterator[bytes]` here as well?

Reply:
done